### ISSUE TITLE
How-To request

### ISSUE THREAD

**Issue by Austin-Williams** [neocogent/sqlchain#1]
Could you include ELI5, step by step instructions for getting this up-and-running on a VPS, and put it in the README?

Like, suppose we just rented a fresh VPS from Amazon and want to get this up and running. Could you put together a quick how-to?

Thank you!

**Comment by neocogent** [comment]
Yes. I will do this but it's not ready yet for more than "playing around with". I'm going to setup up a demo server either on EC2 or a VPS and doing that will let me test the detailed steps and what dependencies are involved. Then I'll revise my readme to be more of a guide rather than just a description.

At the moment if you wanted to test it you would clone the github and edit the cfg dict values at top of each file to suit your system (later I'll blank these out so they need to be set via cmd line). Off the top of my head now I think you'll need the following python modules (use pip install),

python-bitcoinrpc
gevent
gevent-websocket
python-daemon

and mysql-server and python-mysqldb packages if not already installed. I'll write this up better before too long and likely create a package that handles most of this stuff.

Before running **sqlchaind** you would want to create the mysql database using the sqlchain.sql file and probably add a user with access rights. eg.

( from this tutorial - https://www.digitalocean.com/community/tutorials/how-to-create-a-new-user-and-grant-permissions-in-mysql )

```
mysql -u root -p

CREATE USER 'btc'@'localhost' IDENTIFIED BY 'your_password';
GRANT ALL PRIVILEGES ON 'bitcoin.*' TO 'btc'@'localhost';
FLUSH PRIVILEGES;
```

and to create the database 'bitcoin' within mysql,

```
\. etc/sqlchain.sql
```

or from the cmd line,

```
mysql -u root -p <cat etc/sqlchain.sql
```

Add your user,pwd as cfg values in sqlchaind. Also, add the correct rpc user,pwd values. Then you can try,

```
./sqlchaind --debug
```

which will log to screen and should show it adding blocks starting from 0. Use Ctrl-C to stop safely. It can run as a daemon with logging to file but I'd start like this for now so you can easily see what it's doing. 

Currently, you will need at least a partially sync'd blockchain. If it's still syncing then it's way too unresponsive for sqlchain to catch up and you'd get rpc timeouts probably, but if you want to test before fully sync'd then you can start bitcoind with -connect=192.168.5.8 (some non-existing ip) and it will run without being able to sync (which allows it to handle rpc requests quickly).

I have just added code for syncing sqlchain while bitcoind is still syncing (save time and part of my zero footprint ideal) but it's very new and a bit tricky to use. It reads blocks directly form the blk*.dat files and so can build the db even if bitcoind is busy (just as quickly as bitcoind). Also, at this time I have not tested with syncing to a pruning node but there is pruning aware code.

After sqlchaind has got a useful number of blocks then you can try running (edit cfg first),

```
./sqlchain-api --debug
```

which will start the API layer. You can open the demo web page at localhost:8085 or whatever you set in cfg variable. And test (the demo has api examples in the 130,000 block range so some may give null results until you have that many blocks). The demo web page lists all the api calls and lets you see actual results passed back by json (jQuery .get calls).

Anyway, this is just brief off the top my head. I will be adding a better guide later when I have sync'd my node fully and taken notes on a fresh install.

**Comment by neocogent** [comment]
I've now finally created a sqlchain-init script to make setup easy and added an Install Guide to the repo. Sorry it took so long - 1.5 years! I was busy on other work and now I'm doing more work on this - including many major fixes and better testing with current bitcoind.
